### ISSUE TITLE
Rubocop errors on class which includes regex with braces and percents.

### ISSUE THREAD

**Issue by jcavalieri** [bbatsov/rubocop#5817]
Rubocop errors on class which includes scan regex with braces and percents.

There is another issue submitted related to this: https://github.com/bbatsov/rubocop/issues/5815

However, I don't know if the above issue is directly related to this issue.

--------

## Expected behavior

Rubocop should be able to analyze the following class:
https://gist.github.com/jcavalieri/1b2374c673fbbe9f80ae004e79ad22d8

Please note with the above there is a mention of another false positive issue, however, this is a separate issue.

## Actual behavior

Rubocop gets an error:
```
62 files inspected, no offenses detected

1 error occurred:
An error occurred while Lint/RedundantWithObject cop was inspecting /app/app/validators/template_validator.rb:15:6.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
Mention the following information in the issue report:
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

## Steps to reproduce the problem

`rubocop -ad`

## RuboCop version

```
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

Also fails with 0.55.0

**Comment by jcavalieri** [comment]
Closing this, because it turned out there was an error in my code.

I think this might have been failing because I actually forgot to pass in an object with each_with_object.

Once I fixed this, this issue went away.
